packages/core/test/combinator/slice/index-test.ts

```diff
@@ -27,21 +27,21 @@ describe('slice', function () {
     })
 
     it('should narrow when second slice is smaller', function () {
-      const s = slice(1, 5, slice(1, 10, now('')))
+      const s = slice(1, 5, slice(1, 10, now(''))) as Slice<string>
```

Comment:
I had to stare at this test for a minute and think about why this cast is necessary.  Here are a few options that may help to make this more obvious:

1. move the cast to the next line where `.bounds` is accessed, e.g. `(s as Slice<string>).bounds`, _or_
2. add a comment about why the cast is needed (to access `.bounds`), _or_
3. both 1 and 2 😄 

Thoughts?

Reply:
Yeah, the cast is needed only to access `Slice`-specific fields because `slice` combinator defines its return type as `Stream` explicitly. As the test is all about `Slice` I added the cast early to leave a room for additional assertions in the future. I think a comment near the cast in construction would be enough

Reply:
👍 for adding a comment